```
plugins/soundsourcem4a/soundsourcem4a.cpp:242:19: error: use of deleted function ‘mixxx::AudioSourceConfig& mixxx::AudioSourceConfig::operator=(const mixxx::AudioSourceConfig&)’
     m_audioSrcCfg = audioSrcCfg;
                   ^
In file included from src/sources/soundsource.h:5:0,
                 from src/sources/soundsourceprovider.h:8,
                 from src/sources/soundsourcepluginapi.h:30,
                 from src/sources/soundsourceplugin.h:4,
                 from plugins/soundsourcem4a/soundsourcem4a.h:4,
                 from plugins/soundsourcem4a/soundsourcem4a.cpp:1:
src/sources/audiosource.h:218:7: note: ‘mixxx::AudioSourceConfig& mixxx::AudioSourceConfig::operator=(const mixxx::AudioSourceConfig&)’ is implicitly deleted because the default definition would be ill-formed:
```

--- a/plugins/soundsourcemediafoundation/soundsourcemediafoundation.cpp
+++ b/plugins/soundsourcemediafoundation/soundsourcemediafoundation.cpp
@@ -173,7 +173,7 @@ SINT SoundSourceMediaFoundation::seekSampleFrame(
     // sample accurate decoding. The decoder needs to decode
     // some frames in advance to produce the same result at
     // each position in the stream.
-    SINT seekIndex = std::max(SINT(frameIndex - kNumberOfPrefetchFrames), SINT(0));
+    SINT seekIndex = std::max(SINT(frameIndex - kNumberOfPrefetchFrames), AudioSource::getMinFrameIndex());
 
     LONGLONG seekPos = m_streamUnitConverter.fromFrameIndex(seekIndex);
     DEBUG_ASSERT(seekPos >= 0);

--- a/plugins/soundsourcemediafoundation/soundsourcemediafoundation.cpp
+++ b/plugins/soundsourcemediafoundation/soundsourcemediafoundation.cpp
@@ -194,22 +194,29 @@ SINT SoundSourceMediaFoundation::seekSampleFrame(
         //   "After seeking, the application should call IMFSourceReader::ReadSample
         //    and advance to the desired position.
         SINT skipFramesCount = frameIndex - seekIndex;
-        // We need to fetch at least 1 sample from the reader to obtain the
-        // current position!
-        DEBUG_ASSERT(skipFramesCount > 0);
-        skipSampleFrames(skipFramesCount);
-        // Now m_currentFrameIndex reflects the actual position of the reader
-        if (m_currentFrameIndex < frameIndex) {
-            // Skip more samples if frameIndex has not yet been reached
-            skipSampleFrames(frameIndex - m_currentFrameIndex);
-        }
-        if (m_currentFrameIndex != frameIndex) {
-            qWarning() << kLogPreamble
-                    << "Seek to frame"
-                    << frameIndex
-                    << "failed";
-            // Jump to end of stream (= invalidate current position)
-            m_currentFrameIndex = getMaxFrameIndex();
+        if (skipFramesCount > 0) {
+            // We need to fetch at least 1 sample from the reader to obtain the
+            // current position!
+            skipSampleFrames(skipFramesCount);
+            // Now m_currentFrameIndex reflects the actual position of the reader
+            if (m_currentFrameIndex < frameIndex) {
+                // Skip more samples if frameIndex has not yet been reached
+                skipSampleFrames(frameIndex - m_currentFrameIndex);
+            }
+            if (m_currentFrameIndex != frameIndex) {
+                qWarning() << kLogPreamble
+                        << "Seek to frame"
+                        << frameIndex
+                        << "failed";
+                // Jump to end of stream (= invalidate current position)
+                m_currentFrameIndex = getMaxFrameIndex();
+            }
+        } else {
+            // We are at the beginning of the stream and don't need
+            // to skip any frames. Calling IMFSourceReader::ReadSample
+            // is not necessary in this special case.
+            DEBUG_ASSERT(frameIndex == AudioSource::getMinFrameIndex());
+            m_currentFrameIndex = frameIndex;
         }
     } else {
         qWarning() << kLogPreamble

--- a/plugins/soundsourcemediafoundation/soundsourcemediafoundation.h
+++ b/plugins/soundsourcemediafoundation/soundsourcemediafoundation.h
@@ -32,12 +32,13 @@ class StreamUnitConverter final {
         // Used for seeking, so we need to round down to hit the
         // corresponding stream unit where the given stream unit
         // starts
-        return floor(frameIndex * m_streamUnitsPerFrame);
+        return floor((frameIndex - AudioSource::getMinFrameIndex()) * m_streamUnitsPerFrame);
     }
 
     SINT toFrameIndex(LONGLONG streamPos) const {
         // NOTE(uklotzde): Add m_toFrameIndexBias to account for rounding errors
-        return floor((streamPos + m_toFrameIndexBias) / m_streamUnitsPerFrame);
+        return AudioSource::getMinFrameIndex() +
+                static_cast<SINT>(floor((streamPos + m_toFrameIndexBias) / m_streamUnitsPerFrame));
     }
 
   private:

--- a/src/util/audiosignal.h
+++ b/src/util/audiosignal.h
@@ -71,14 +71,7 @@ public:
         DEBUG_ASSERT(kSamplingRateZero <= m_samplingRate); // unsigned value
     }
 
-    // all-default memory management
-    AudioSignal(const AudioSignal&) = default;
     virtual ~AudioSignal() = default;
-// Visual Studio does not support default generated move constructors yet
-#if !defined(_MSC_VER) || _MSC_VER > 1900
-    AudioSignal(AudioSignal&&) = default;
-    AudioSignal& operator=(AudioSignal&&) = default;
-#endif
 
     // Returns the ordering of samples in contiguous buffers
     SampleLayout getSampleLayout() const {

